File: Android/src/jp/co/riso/android/text/IpAddressFilter.java
/*
 * Copyright (c) 2014 RISO, Inc. All rights reserved.
 *
 * IpAddressFilter.java
 * SmartDeviceApp
 * Created by: a-LINK Group
 */

package jp.co.riso.android.text;

import java.util.HashSet;
import java.util.Set;

import android.text.InputFilter;
import android.text.Spanned;

/**
 * EditText filter for IP Address characters
 */
public class IpAddressFilter implements InputFilter {
    private Set<Character> mCharSet;
    
    /**
     * Constructor
     */
    public IpAddressFilter() {
        String validChars = "1234567890abcdefABCDEF.:";
        mCharSet = new HashSet<Character>();
        for (int i = 0; i < validChars.length(); i++) {
            mCharSet.add(validChars.charAt(i));
        }
    }
    
    /** {@inheritDoc} */
    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        for (int i = start; i < end; i++) {
            if (!mCharSet.contains(source.charAt(i))) {
                return "";
            }
        }
        return null;
    }
}

File: Android/app/src/main/kotlin/jp/co/riso/android/text/IpAddressFilter.kt
/*
 * Copyright (c) 2022 RISO, Inc. All rights reserved.
 *
 * IpAddressFilter.kt
 * SmartDeviceApp
 * Created by: a-LINK Group
 */
package jp.co.riso.android.text

import android.text.InputFilter
import android.text.Spanned
import java.util.HashSet

/**
 * @class IpAddressFilter
 *
 * @brief EditText filter for IP Address characters
 */
class IpAddressFilter : InputFilter {
    private val _charSet: MutableSet<Char>
    override fun filter(
        source: CharSequence,
        start: Int,
        end: Int,
        dest: Spanned,
        dstart: Int,
        dend: Int
    ): CharSequence? {
        for (i in start until end) {
            if (!_charSet.contains(source[i])) {
                return ""
            }
        }
        return null
    }

    init {
        val validChars = "1234567890abcdefABCDEF.:"
        _charSet = HashSet()
        for (element in validChars) {
            _charSet.add(element)
        }
    }
}